File: biopython_rewrite/split_fasta.py
#!/usr/bin/env python3
# Split multi-FASTA into files
import sys, os
from Bio import SeqIO
if len(sys.argv)<2:
    print('Usage: split_fasta.py multi.fasta'); sys.exit(1)
for rec in SeqIO.parse(sys.argv[1],'fasta'):
    out = rec.id + '.fasta'
    SeqIO.write(rec, out, 'fasta')

File: legacy_perl/split_fasta.pl
#!/usr/bin/env perl
# Split multi-FASTA into individual files
use strict; use warnings; use Bio::SeqIO;
my $in = shift or die "Usage: $0 multi.fasta\n";
my $seqio = Bio::SeqIO->new(-file=>$in,-format=>'fasta');
while(my $r=$seqio->next_seq){
    my $id = $r->display_id;
    my $out = $id . '.fasta';
    my $o = Bio::SeqIO->new(-file=>">$out", -format=>'fasta');
    $o->write_seq($r);
}
